testsuite/MDAnalysisTests/formats/test_libmdaxdr.py

```diff
@@ -21,376 +21,469 @@
 import numpy as np
 
+from numpy.testing import (assert_almost_equal, assert_array_almost_equal,
+                           assert_array_equal)
 
-class XDRFormatBaseTest(object):
+from MDAnalysis.lib.formats.libmdaxdr import TRRFile, XTCFile
 
-    def test_n_atoms(self):
-        f = self.xdrfile(self.single_frame)
-        assert_equal(f.n_atoms, 10)
+from MDAnalysisTests.datafiles import TRR_multi_frame, XTC_multi_frame
 
-    @raises(IOError)
-    def test_raise_not_existing(self):
-        self.xdrfile('foo')
+import pytest
 
-    @raises(IOError)
-    def test_open_wrong_mode(self):
-        self.xdrfile('foo', 'e')
+XTC_OFFSETS = np.array([0, 104, 208, 312, 416, 520, 624, 728, 832, 936])
+TRR_OFFSETS = np.array([0, 480, 960, 1440, 1920, 2400, 2880, 3360, 3840, 4320])
 
-    @raises(IOError)
-    def test_over_seek(self):
-        with self.xdrfile(self.multi_frame) as f:
-            f.seek(100)
 
-    @raises(IOError)
-    @run_in_tempdir()
-    def test_read_write_mode_file(self):
-        with self.xdrfile('foo', 'w') as f:
-            f.read()
-
-    @raises(IOError)
-    def test_read_closed(self):
-        f = self.xdrfile(self.multi_frame)
-        f.close()
-        f.read()
+@pytest.fixture
+def xtc():
+    with XTCFile(XTC_multi_frame) as f:
+        yield f
 
-    def test_iter(self):
-        with self.xdrfile(self.multi_frame) as f:
-            for frame in f:
-                pass
 
-    def test_tell(self):
-        f = self.xdrfile(self.multi_frame)
-        assert_equal(f.tell(), 0)
-        for i, frame in enumerate(f):
-            assert_equal(f.tell(), i + 1)
-
-    def test_seek(self):
-        f = self.xdrfile(self.multi_frame)
-        f.seek(4)
-        assert_equal(f.tell(), 4)
-
-    def test_offset(self):
-        f = self.xdrfile(self.multi_frame)
-        assert_array_equal(f.offsets, self.offsets)
-
-    @raises(IOError)
-    def test_set_offsets(self):
-        f = self.xdrfile(self.multi_frame)
-        f.set_offsets(np.arange(len(self.offsets)))
-        assert_array_equal(f.offsets, np.arange(len(self.offsets)))
-        f.seek(6)
-        f.read()
-
-    @raises(OverflowError)
-    def test_seek_overflow(self):
-        f = self.xdrfile(self.multi_frame)
-        f._bytes_seek(2**65)
-
-    def test_seek_tell_all(self):
-        f = self.xdrfile(self.multi_frame)
-        for frame, offset in enumerate(self.offsets):
-            f.seek(frame)
-            assert_equal(f._bytes_tell(), offset)
-
-    def test_seek_tell_all_bytes(self):
-        f = self.xdrfile(self.multi_frame)
-        for offset in self.offsets:
-            f._bytes_seek(offset)
-            assert_equal(f._bytes_tell(), offset)
-
-    def test_seek_tell_largefile(self):
+@pytest.fixture
+def trr():
+    with TRRFile(TRR_multi_frame) as f:
+        yield f
+
+
+@pytest.mark.parametrize('fname, xdr', ((XTC_multi_frame, XTCFile),
+                                        (TRR_multi_frame, TRRFile)))
+class TestCommonAPI:
```

Comment:
@richardjgowers is that what you have in common with using classes to parametrize common tests?